### ISSUE TITLE
Use pandas categorical dtype encoding where applicable 

### ISSUE THREAD

**Issue by gsheni** [alteryx/evalml#2360]
- Woodwork uses the `category` dtype where applicable for certain Logical Types.
- One of the [benefits](https://pandas.pydata.org/pandas-docs/stable/user_guide/categorical.html) of using this pandas dtype is that pandas will automatically encode the column and provide you with a mapping. 
- EvalML is currently not using this functionality and may benefit from it (possible in the **OneHotEncoder**?)

**Comment by dsherry** [comment]
Yep. Question: does our one-hot encoder currently output `category` type? I thought it did.

Filing as spike--if we're not using the right type, let's use the right type :)

**Comment by angela97lin** [comment]
More context for whoever picks this up: https://ayx-innovation-labs.slack.com/archives/CPYLCJR9Q/p1623272174251500
